ui/pages/bridge/prepare/prepare-bridge-page.tsx

```diff
@@ -180,15 +180,50 @@ const PrepareBridgePage = () => {
   );
 
   const wasTxDeclined = useSelector(getWasTxDeclined);
-  // If latest quote is expired and user has sufficient balance
-  // set activeQuote to undefined to hide stale quotes but keep inputs filled
-  const activeQuote =
-    isQuoteExpired &&
-    (!quoteRequest.insufficientBal ||
-      // insufficientBal is always true for solana
-      (fromChain && isSolanaChainId(fromChain.chainId)))
-      ? undefined
-      : activeQuote_;
+
+  // A quote is considered invalid if it is expired or if its destination token/chain
+  // no longer matches the current user selection. This prevents a race condition
+  // where a stale quote (e.g., for SOL→USDC) could be used after the user has
+  // already switched the destination token (e.g., to SOL→AI).
+  const isQuoteExpiredOrInvalid = () => {
+    // 1. Expired quotes should be ignored
+    if (
+      isQuoteExpired &&
+      (!quoteRequest.insufficientBal ||
+        // `insufficientBal` is always true for Solana
+        (fromChain && isSolanaChainId(fromChain.chainId)))
+    ) {
+      return true;
+    }
+
+    // 2. If the quote's destination asset or chain no longer matches the UI
+    //    selection, treat it as invalid.
+    if (activeQuote_ && toToken) {
+      const quoteDestAddress = (
+        activeQuote_.quote?.destAsset?.address || zeroAddress()
+      ).toLowerCase();
+      const selectedDestAddress = (
+        toToken.address || zeroAddress()
+      ).toLowerCase();
+      const quoteDestChainIdCaip = activeQuote_.quote?.destChainId
+        ? formatChainIdToCaip(activeQuote_.quote.destChainId)
+        : '';
+      const selectedDestChainIdCaip = toChain?.chainId
+        ? formatChainIdToCaip(toChain.chainId)
+        : '';
+
+      if (
+        quoteDestAddress !== selectedDestAddress ||
+        quoteDestChainIdCaip !== selectedDestChainIdCaip
+      ) {
+        return true;
+      }
+    }
```

Comment:
Minor: this can be simplified to this
```suggestion
      return !(
        quoteDestAddress === selectedDestAddress &&
        quoteDestChainIdCaip === selectedDestChainIdCaip
      )
    }
```